azkustodata/kusto.go

```diff
@@ -172,6 +177,86 @@ func (c *Client) Query(ctx context.Context, db string, query Statement, options
 	return iter, nil
 }
 
+func (c *Client) newV1(
+	op errors.Op, call int,
+	ctx context.Context, db string, kqlQuery Statement, options ...QueryOption) (v1.Dataset, error) {
+	ctx, cancel := contextSetup(ctx)
+
+	opQuery := op
+	opts, err := setQueryOptions(ctx, opQuery, kqlQuery, call, options...)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := c.getConn(mgmtCall, connOptions{queryOptions: opts})
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := conn.rawQuery(ctx, db, kqlQuery, opts)
+
+	if err != nil {
+		cancel()
+		return nil, err
+	}
+
+	return v1.NewDatasetFromReader(ctx, opQuery, res)
+}
+
+func (c *Client) MgmtNew(ctx context.Context, db string, kqlQuery Statement, options ...QueryOption) (v1.Dataset, error) {
+	return c.newV1(errors.OpMgmt, mgmtCall, ctx, db, kqlQuery, options...)
+}
+
+func (c *Client) QueryNew(ctx context.Context, db string, kqlQuery Statement, options ...QueryOption) (queryv2.FullDataset, error) {
+	_, res, err := c.newV2(ctx, db, kqlQuery, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return queryv2.NewFullDataSet(ctx, res)
+}
+
+func (c *Client) IterativeQuery(ctx context.Context, db string, kqlQuery Statement, options ...QueryOption) (queryv2.IterativeDataset, error) {
+	opts, res, err := c.newV2(ctx, db, kqlQuery, options)
+	if err != nil {
+		return nil, err
+	}
+
+	capacity := queryv2.DefaultFrameCapacity
+	if opts.v2FrameCapacity != -1 {
+		capacity = opts.v2FrameCapacity
+	}
+
+	return queryv2.NewStreamingDataSet(ctx, res, capacity)
+}
+
+func (c *Client) newV2(ctx context.Context, db string, kqlQuery Statement, options []QueryOption) (*queryOptions, io.ReadCloser, error) {
+	ctx, cancel := contextSetup(ctx)
+
+	options = append(options, V2NewlinesBetweenFrames())
+	options = append(options, V2FragmentPrimaryTables())
+	options = append(options, ResultsErrorReportingPlacement(ResultsErrorReportingPlacementEndOfTable))
+
+	opQuery := errors.OpQuery
+	opts, err := setQueryOptions(ctx, opQuery, kqlQuery, queryCall, options...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	conn, err := c.getConn(queryCall, connOptions{queryOptions: opts})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	res, err := conn.rawQuery(ctx, db, kqlQuery, opts)
+
+	if err != nil {
+		cancel()
+		return nil, nil, err
+	}
+	return opts, res, nil
```

Comment:
Where's the whole iterative dataset reader?

Reply:
all of the "major" code is in query.v2